### PR TITLE
🚧 Products v3: viewing, searching,  filtering & pagination

### LINKED ISSUE #10679: [BUU] As an enterprise user, I want to know if I have products in my catalogue
## Context
This is the first issue that uses the new tech stack and new styling and components. We are going to set up the foundations of the new redesigned page, creating the first few elements (title, button, text for content), and assessing page loading. 
Since this issue covers error states as well, but there is a redesign of the illustrations in progress, we'll just include some placeholder content and [update once the imagery is ready](https://github.com/openfoodfoundation/openfoodnetwork/issues/10680).
We are not going to build the table itself in this card, but just the loading, empty and error states, and a placeholder state telling the user that they have products.

## Description

**- As an:** enterprise user 
**- On page:** `/admin/products`
**- I want to be able to:** know if I have products in my catalogue
**- So that:** I can decide if I need to create or import new ones or not

## Acceptance Criteria & Tests

**Scenario 1: Loading the page for the first time**
Given that I am logged in
When I land to `/admin/products`
Then the page starts loading my catalogue
And I see a loading indicator

Note: we will first build a version of the table loading that includes a simple loading indicator. Once we know more about how loading data is going to behave, we might potentially upgrade this to something more advanced.

Mockup > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A218&t=VkGk4jMtqb93Kksj-1

**Scenario 2: Successful loading with products**
Given that I have products in my catalogue
And I am in `/admin/products`
And the page is loading my catalogue
When the catalogue loads successfully
Then I see a message telling me that I have products in my catalogue

Mockup > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A631&t=VkGk4jMtqb93Kksj-1

**Scenario 3: Creating a new product from catalogue with products**
Given that I have products in my catalogue
And I am in `/admin/products`
And the page has loaded my catalogue successfully
When I click on the button to create a new product
Then I go to `/admin/products/new`

Note: this button should be shown only if I have products in my catalogue, not for empty and error states.

**Scenario 4: Successful loading without products**
Given that I don't have products
And I am in `/admin/products`
And the page is loading my catalogue
When the page loads my catalogue successfully
Then I see a message telling me that I don't have products in my catalogue

Mockup > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A227&t=VkGk4jMtqb93Kksj-1

**Scenario 5: Creating a new product from catalogue without products**
Given that I don't have products in my catalogue
And I am in `/admin/products`
And the page has loaded my (empty) catalogue successfully
When I click on the button to create a new product
Then I go to `/admin/products/new`

**Scenario 6: Importing products from catalogue without products**
Given that I don't have products in my catalogue
And I am in `/admin/products`
And the page has loaded my (empty) catalogue successfully
When I click on the button to import products
Then I go to `/admin/product_import`

**Scenario 7: Failed loading - no network**
Given that I don't have network connection
And I am in `/admin/products`
And the page is trying to load my catalogue
When the page stops trying to load my catalogue
Then I see a message telling me that I don't have network connection

Note: this scenario might be needed in case we have the ability to refresh the table without fully reloading the page. this scenario might not be necessary if we just go with full page load and no network browser behaviour.

Mockup > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A235&t=VkGk4jMtqb93Kksj-1

**Scenario 8: No network - refresh**
Given that I am in `/admin/products`
And the page has failed to load due to no network connection
When I click on the button to refresh
Then the page reloads my catalogue
And I see a loading indicator

Note: same as previous scenario.

**Scenario 9: Failed loading - catch'all**
Given that something is not working properly in the background
And I am in `/admin/products`
And the page is trying to load my catalogue
When the page stops trying to load my catalogue
Then I see a message informing me that something is not working.

Mockup > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A242&t=VkGk4jMtqb93Kksj-1

**Scenario 10: Catch'all - refresh**
Given that I am in `/admin/products`
And the page has failed to load my catalogue due to something not working properly in the background
When I click on the button to refresh
Then the page reloads
And I see a loading indicator

## Design specs

The whole set of screens is available here
https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?node-id=7%3A183&t=VkGk4jMtqb93Kksj-1

**New components and styles**

| Component | Example |
| --- | --- |
| Page title - H1 | Screenshot 2023-04-11 at 5 13 21 pm |
| Body text - regular | Screenshot 2023-04-11 at 5 14 32 pm
| Body text - semibold | Screenshot 2023-04-11 at 5 16 51 pm |
| Primary button with left icon | Screenshot 2023-04-11 at 5 14 39 pm |
| Secondary button with left icon | Screenshot 2023-04-11 at 5 14 53 pm |
| Page header - title only | Screenshot 2023-04-11 at 5 05 29 pm |
| Page header - with primary action  | Screenshot 2023-04-11 at 5 06 10 pm |
| Spinning loading indicator with message | Screenshot 2023-04-11 at 5 14 19 pm |

### LINKED ISSUE #10877: [BUU] Pagination for product list
## Context
This issue enhances the display of the product list table #10694.

## Out of scope
- truncate pagination links #10878 

## Description

**- As an:** enterprise user 
**- On page:** `/admin/products`
**- I want to be able to:** move between pages in my catalogue
**- So that:** I can load my products quickly and navigate between them

## Acceptance Criteria & Tests

**Scenario 1: Less than 25 products**
Given I have less than 25 products in my catalogue
And I am in `/admin/products`
And the page is loading my catalogue
When my catalogue has been loaded successfully
Then I don't see any pagination

**Scenario 2: More than 25 products**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
And the page is loading my catalogue
When my catalogue has been loaded successfully
Then I see the pagination

**Scenario 3: Clicking a page link**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
When I click on a page number
Then the table loads the products for the page I have clicked on

**Scenario 4: Disabled previous button**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
And the page is loading the first 25 products in my catalogue
When my catalogue has been loaded successfully
Then the button to go to the previous page is disabled 

**Scenario 5: Disabled next button**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
And the page is loading the last 25 (or less) products in my catalogue
When my catalogue has been loaded successfully
Then the button to go to the next page is disabled

**Scenario 6: Next button**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
And I am not in the last page of the catalogue
When I click on the next button in pagination
Then the catalogue loads the following 25 products in the list

**Scenario 7: Previous button**
Given I have more than 25 products in my catalogue
And I am in `/admin/products`
And I am not in the first page of the catalogue
When I click on the previous button in pagination
Then the catalogue loads the previous 25 products in the list

## Design specs
The screen for specs is here
https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?type=design&node-id=119-734

**New components and styles**

| Component | Example |
| --- | --- |
| Square action button | Screenshot 2023-05-29 at 3 17 22 pm |

### LINKED ISSUE #11054: [BUU] Change the number of products shown in my catalogue
## Context
This issue enhances the display of the product list table #10694.

## Out of scope
- filter by producer or category #11057
- search for products https://github.com/openfoodfoundation/openfoodnetwork/issues/11056

## Description

**- As an:** enterprise user 
**- On page:** `/admin/products`
**- I want to be able to:** choose how many products to see
**- So that:** I can scan through as many products as I want

## Acceptance Criteria & Tests

**Scenario 1: 15 or less products - NICE TO HAVE**
Given that I am in `/admin/products`
And I have 15 or less products
When the page has loaded my products 
Then I see all of my products
And I don't see a picker to change how many products are displayed

**Scenario 2: More than 15**
Given that I am in `/admin/products`
And I have more than 15 products
When the page has loaded my products 
Then I see the first 15 products of my catalogue
And I see a picker to change how many products are displayed

**Scenario 3: Change n. of product to display**
Given I have more than 15 products
And the page has loaded my first 15 products
When I select a number different than 15 products to display
Then the table refreshes
And I see the number of products I have selected

Available options:
- 15 products (default)
- 50 products
- 100 products

**Scenario 4: Pagination**
Given I have selected a number of products to be displayed different from 15
And there are multiple pages of products available
When I move between pages of products
Then I always see the number of products that I have initially selected

**Scenario 5: Searching and filtering**
Given I have selected a number of products to be displayed different from 15
When I search for a product (and/or [filter](https://github.com/openfoodfoundation/openfoodnetwork/issues/11057))
Then I always see the number of products that I have initially selected

## Design specs
A prototype is available here > https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?page-id=373%3A2076&type=design&node-id=373-2077&viewport=625%2C637%2C0.62&t=1a6HPQf2tS4LA7rY-1&scaling=min-zoom&starting-point-node-id=373%3A2077&mode=design

Note: the re-loading behaviour is a concept work in progress for now - implementation might be different depending on what's more feasible.

Figma screens are available here > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?type=design&node-id=373%3A2076&mode=design&t=PlajwLujJOXIZIjx-1

**New components and styles**

| Component | Example |
| --- | --- |
| Regular simple dropdown closed | Screenshot 2023-07-12 at 14 53 12 |
| Regular simple dropdown open - with current selection | Screenshot 2023-07-12 at 14 55 42|

### LINKED ISSUE #11056: [BUU] Search for a product
## Context
This issue enhances the display of the product list table #10694.

## Out of scope
- filter by producer or category #11057

## Description

**- As an:** enterprise user 
**- On page:** `/admin/products`
**- I want to be able to:** search for products in my catalogue
**- So that:** I can find the product I am looking for

## Acceptance Criteria & Tests

**Scenario 1: Active search field**
Given that I am in `/admin/products`
And the page has already loaded my products
When I select the search field
Then the search field becomes active
And I can type my text query in it

**Scenario 2: Trigger search**
Given I have typed my query in the search field
And the query is valid
When I trigger searching (keyboard enter or pressing the button)
Then the system searches for products containing that query
And the table is replaced by a loading indicator

**Notes to explore: pasting text, enabling emojis/special characters, what fields is the system searching in (only product title? variants titles? product description? etc..)**

**Scenario 3: No results**
Given I have triggered a search in my catalogue
And there are no products matching that query
When the system responds to my request
Then I see message telling me that there are no results for my search

**Scenario 4: Clear search on no results**
Given there are no results for my search
When I clear my search
Then I see my catalogue with all the products

**Scenario 5: Search with results**
Given I have triggered a search in my catalogue
And there are products matching that query
When the system responds to my request
Then I see the list of products that match that query

**Scenario 6: Search with results - multiple pages**
Given I have triggered a search in my catalogue
And there are more than 25 products matching that query
When the system responds to my request
Then I see the list of products that match that query
And pagination at the bottom of the list

**Scenario 7: Move to next page**
Given there are more than 25 products in my search results
When I click on the next page button in pagination
Then the catalogue loads the following 25 products in the list of search results for my query

## Design specs
A prototype is available here > https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?page-id=346%3A903&type=design&node-id=346-1000&viewport=657%2C461%2C0.16&t=gG17pR0ejGmiUT88-1&scaling=min-zoom&starting-point-node-id=346%3A1000&mode=design
Figma screens are available here > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?type=design&node-id=346%3A903&mode=design&t=W6f7NFDHptqN9Fh7-1

**New components and styles**

| Component | Example |
| --- | --- |
| Relaxed text field | Screenshot 2023-07-05 at 14 55 44 |
|Relaxed secondary button| Screenshot 2023-07-05 at 14 56 06|
|Regular secondary button disruptive|Screenshot 2023-07-05 at 14 56 14|
|Relaxed secondary button disruptive| Screenshot 2023-07-05 at 14 56 22|

### LINKED ISSUE #11057: [BUU] Filter my catalogue
## Context
This issue enhances the display of the product list table #10694.

## Out of scope
- search for products https://github.com/openfoodfoundation/openfoodnetwork/issues/11056

## Description

**- As an:** enterprise user 
**- On page:** `/admin/products`
**- I want to be able to:** filter my products by producer and category
**- So that:** I can find the products I'm looking for

## Acceptance Criteria & Tests

**Scenario 1: Single producer**
Given that I am in `/admin/products`
And my enterprise only has one producers
When the page has loaded my products 
Then I see only a filter for categories (All categories pre-selected)
And I don't see a filter for producers

**Scenario 2: Multiple producers**
Given that I am in `/admin/products`
And my enterprise only has more than one producer
When the page has loaded my products 
Then I see a filter for categories (All categories pre-selected)
And a filter for producers (All producers pre-selected

Note: logic to display producers picker would be the same as tracked in https://github.com/openfoodfoundation/openfoodnetwork/issues/11200 (decoupled from the original product table display)

**Scenario 3: Selecting different option**
Given that I am in `/admin/products`
When I select a different option for Producer or Category
Then the picker shows my new selection
And the table does not refresh

**Scenario 4: Searching within filter**
Given that I am in `/admin/products`
And I have opened a filter
When I search within a filter
Then I see a subset of options based on my search

**Scenario 5: Filtering**
Given that I am in `/admin/products`
And I have selected a different option for Producer or Category
When I search
Then the table refreshes
And shows products that match my filters

**Scenario 6: Pagination**
Given I have filtered my products
And there are more results than what are displayed on the table
When I go to the second page of the results
Then I see more products
And the filter remains as per my selection

## Design specs
A prototype is available here (categories can be clicked on) > https://www.figma.com/proto/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?page-id=378%3A3907&type=design&node-id=378-3908&viewport=546%2C542%2C0.58&t=OOUsawRfEBZzOZVj-1&scaling=min-zoom&starting-point-node-id=378%3A3908&mode=design

Figma screens are available here > https://www.figma.com/file/v1zbrWDZSRd3Nqoe0SJ2Sm/Engineering-Delivery?type=design&node-id=378%3A3907&mode=design&t=TgZ3irJGBg2CvOuz-1

**New components and styles**

| Component | Example |
| --- | --- |
| Relaxed dropdown closed | Screenshot 2023-07-12 at 16 40 17 |
| Relaxed dropdown open - with inline search | Screenshot 2023-07-12 at 16 40 59|

### LINKED ISSUE #11268: [BUU] Allow to activate feature toggle for users outside of superadmin + one actor only

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

This PR introduce the first bunch of features for the new products V3 page, under the `admin_style_v3` feature toggle.

Capture d’écran 2023-07-11 à 09 31 40
Capture d’écran 2023-07-11 à 09 31 44
Capture d’écran 2023-07-11 à 09 31 55
Capture d’écran 2023-07-11 à 09 32 12

Includes:
- #10694
- #11005
- Closes #10679
- Closes #10877
- Closes #11054 
- Closes #11056 
- Closes #11057 
- Closes #11268

#### What should we test?
There are extensive acceptance criteria on each of the issues listed above (although they behave as you'd expect). The first two had been merged already but haven't yet been tested. Please review each of these, and provide any feedback you might have.

#### Release notes

Changelog Category: User facing changes, under feature toggle

### CODE REVIEW

File: spec/system/admin/products_v3/products_spec.rb
Comment (dacook) [inline spec/system/admin/products_v3/products_spec.rb:117]: We could also check that the correct product is showing (we'll need to give the product a name).

File: app/reflexes/products_reflex.rb
Comment (dacook) [inline app/reflexes/products_reflex.rb:39]: Why not just hook these up directly with `before_reflex`?

File: app/reflexes/products_reflex.rb
Comment (dacook) [inline app/reflexes/products_reflex.rb:47]: It's a shame we need to manage these this way.. surely there is a better way? Can the JS attached to the "search" button somehow submit only the form params in the request?

File: app/webpacker/css/admin_v3/all.scss
Comment (dacook) [inline app/webpacker/css/admin_v3/all.scss:54]: I think it would be best to stick to the rule of replacing the old file entirely

```suggestion
```

**Review by dacook — CHANGES_REQUESTED** [review]
Phew, epic job! 
As you're aware, it was hard to review as there's a lot of commits and changes. I've got some suggestions, and various comments. I would have gone ahead and tried these out but I'm afraid I've run out of time today.

I also think it will help to pull out my unrelated commits, particularly because one has an effect outside of the feature toggle that will be worth a quick test (I don't want to break the admin interface again!). I'll go ahead with that,  and update this branch (and squash a few fixup commits).

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:72]: Because this change the URL, and if we change the URL before the cable_ready.replace this causes an error (cant' remember but something like : "can't update DOM since URL has changed")

Comment (jibees) [reply app/reflexes/products_reflex.rb:72]: the exact error is
```
stimulus_reflex.js:690 Reflex failed due to mismatched URL.
```

File: app/webpacker/css/admin/products_v3.scss
Comment (jibees) [reply app/webpacker/css/admin/products_v3.scss:106]: Yep, I thought about this. Let's do this the next time we need this?

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:72]: Oh.. interesting! In that case using `broadcast` (now) would probably be safer.

It seems like this should be easier to solve though. Surely this pattern is a normal one. I would think it's equivalent to using `<form method="get">`.

File: spec/system/admin/products_v3/products_spec.rb
Comment (jibees) [reply spec/system/admin/products_v3/products_spec.rb:8]: Thanks for your attention

File: spec/system/admin/products_v3/products_spec.rb
Comment (jibees) [reply spec/system/admin/products_v3/products_spec.rb:48]: Right!

File: app/webpacker/controllers/products_controller.js
Comment (jibees) [reply app/webpacker/controllers/products_controller.js:35]: > instead of broadcasting named events like "show-loading", we're more directly sending the showLoading message to the controller.

Exact!

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb]: 👌

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:39]: Yes, that's probably easier to understand ;)

File: app/webpacker/css/admin_v3/all.scss
Comment (jibees) [reply app/webpacker/css/admin_v3/all.scss:54]: Yep, I forget. Thanks.

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:42]: I can't make it work this way either. Would you like to give a try? Maybe this needs a fresh brain

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb]: > I think this would continue to add the same parameters to the url each time you view the next page.

Nop, because `request.original_url` is `http://localhost:3000/admin/products_v3` even is has params.

File: spec/system/admin/products_v3/products_spec.rb
Comment (jibees) [reply spec/system/admin/products_v3/products_spec.rb:111]: Yes!

**Comment by jibees** [comment]
Hi @dacook ! Thanks for this deep review. All the thumbs-up-ed comment you've made have been addressed.

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb]: Oh! 👍

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:67]: I just realised that my suggestion would probably still create the error you mentioned, so I'll leave this.

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:42]: I don't think I have a fresh brain to offer :P
I did a little bit of googling and couldn't find an answer.

**Review by dacook — APPROVED** [review]
Thanks JB, great work.

I just booted it up in my dev environment and it's so beautiful and fast.. 🤩 

I think for a second review, it would be better to have a fresh perspective and **look at the overall diff on the _Files changed_ tab**, rather than review all the commits.

**Comment by dacook** [comment]
I've updated the testing notes to point to the acceptance criteria of each issue. I also re-ordered the list based on the order they were created.

File: app/reflexes/products_reflex.rb
Comment (mkllnk) [inline app/reflexes/products_reflex.rb:14]: I would expect to see the same elements as before, just more or less. We would need the old values though.

```
position = old_page * old_per_page
@page = position / @per_page
@page = @page.clamp(1, @page)
```

File: app/reflexes/products_reflex.rb
Comment (mkllnk) [reply app/reflexes/products_reflex.rb:67]: I don't really know the context. So I'm raising the obvious questions again: Why do we change the URL? Or why don't we use GET to automate that?

File: app/reflexes/products_reflex.rb
Comment (mkllnk) [inline app/reflexes/products_reflex.rb:105]:
```suggestion
      search_key = %w[
        name
        meta_keywords
        # ...
      ].join("_or_") + "_cont"
      query.merge!(search_key => @search_term)
```

This could also be a constant on the variant model.

**Review by mkllnk — APPROVED** [review]
Great work! :muscle: :sparkles:

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:67]: > Why do we change the URL?

To save filters.

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb]: 👌  ff4603ca35

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:14]: i'm not sure to follow you. Would you like to make me a suggestion?

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:67]: > why don't we use GET to automate that?

For example, trigger a normal request with a form submit? 
A form submit would require Rails to rebuild the whole page, which would take slightly longer but that's not really a problem.

But we also want to implement the loading behaviour: show a loading spinner. So if we're going to intercept the basic behaviour, I think we're going to have to also change the URL with JS.

Which brings me to a conclusion: why not update the URL in `productsV3_controller.js` instead?

(caveat: I'm still getting used to all the different options we have here so might not be seeing the whole picture)

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:14]: I think what Maikel's saying is that whenever you change the 'per page' amount, it will reset you back to page 1, which is not intuitive. 

So I think this code suggestion is to select the closest page that includes some/all of the records that were previously on-screen. Is that right @mkllnk ?

Comment (mkllnk) [reply app/reflexes/products_reflex.rb:14]: Exactly, I would see the same first row of the results and the tail of the results is extended or reduced.

Comment (jibees) [reply app/reflexes/products_reflex.rb:14]: > whenever you change the 'per page' amount, it will reset you back to page 1, which is not intuitive.

Oh, ok, I thought it was intuitive ... Ok, let's do this!

> I would see the same first row of the results and the tail of the results is extended or reduced.

Makes sense!

Comment (jibees) [reply app/reflexes/products_reflex.rb:14]: i can't make it work. I've tested the above code for sure, + tested mine.
I don't think it's that easy (specially when `per_page` aren't multiples each others) and I've spent to much time trying to fix it. Maybe open an issue?

File: app/reflexes/products_reflex.rb
Comment (jibees) [reply app/reflexes/products_reflex.rb:67]: > Which brings me to a conclusion: why not update the URL in productsV3_controller.js instead?

We don't have access to the pagination, filtering, ... values, which are handled via the reflex. Actually, I was thinking the opposite, ie. removing `app/webpacker/controllers/products_controller.js` file and only use `app/reflexes/products_reflex.rb` to have one single source of truth.

**Comment by jibees** [comment]
This one has already two approval reviews, but I'd be happy if @mkllnk and @dacook could see my comments and approve (or not). Back in `Code Review` then.

File: app/reflexes/products_reflex.rb
Comment (mkllnk) [reply app/reflexes/products_reflex.rb:14]: Okay, let's leave it for now then.

**Review by mkllnk — APPROVED** [review]

File: app/reflexes/products_reflex.rb
Comment (dacook) [reply app/reflexes/products_reflex.rb:67]: 🤔  
Hmm, maybe we should move on and reconsider this later..

**Comment by jibees** [comment]
No more blocked since this PR will close #11268 (we cherry picked the commit from #11269 )

**Comment by RachL** [comment]
OK there are some stuff to fine tune, but as it's behind a feature toggle and this PR is already monstrously huge, let's unblocked this work! @jibees you can merge 💪

**Comment by dacook** [comment]
Hooray!

Just to confirm, were the eight issues (listed in the description above, now closed) tested to pass?

**Comment by RachL** [comment]
@dacook yes apart from error page display due to fail in the background which I don't know how to reproduce.
Not everything was working perfectly, but new issues will be open. In the future, it would help testers to have 1PR = 1 issue, this was hard to test and keep track :/

**Pull request merged**
